sirbot/slack/store/user.py

```diff
@@ -56,16 +56,23 @@ class UserStore(SlackStore):
     def __init__(self, client, facades, refresh=3600):
         super().__init__(client, facades, refresh)
 
-    async def all(self):
+    async def all(self, fetch=False, deleted=False):
         db = self._facades.get('database')
-        data = await database.__dict__[db.type].user.get_all(db)
-
+        db_data = await database.__dict__[db.type].user.get_all(db)
+        if fetch:
+            fetched_data = await self._client.get_users()
+            for user in fetched_data:
+                await database.__dict__[db.type].user.add_multiple(db, user)
```

Comment:
By seeing the code I realise we don't really need an `add_multiple` at the moment because its taking an `User` object. But you still need to create the object.

```
if fetch:
    users = list()
    fetched_data = await self._client...

    for data in fetched_data:
        user = User(
                   id = ...
                    ...)
         await database.__dict__[db.type].user.add(db, user)
        if deleted:
             users.append(user)
       if not deleted and not user.deleted:
            users.append(user)
else:
    data = await database.__dict__[db.type].user.get_all(db, deleted=deleted)  # We will need to do the SQL for that
    users = [User(....) for raw_data in data]
```